lib/graphiql/rails.rb

```diff
@@ -18,14 +18,44 @@
 
 module GraphiQL
   module Rails
+    class Config
+      attr_accessor :query_params, :initial_query, :headers
+
+      def initialize(query_params:, initial_query:, headers:)
+        @query_params = query_params
+        @initial_query = initial_query
+        @headers = headers
+      end
+
+      def csrf=(_)
+        raise %|
+
+csrf option has been removed, set the header explicitly:
+
+  GraphiQL::Rails.config.headers['X-CSRF-Token'] = ->(context) do
+    context.form_authenticity_token
+  end
```

Comment:
I was going todo that but when I looked at the rest of your code it's so clean so I though you would appreciate a pure approach.

I have added the CRSF token by default and some deprecation warnings, if someone has `csrf = false` on then this will not work but I don't think this would cause anyone any issues.